script.js

```diff
@@ -1,4 +1,149 @@
+// Requirements
+// 1. There are 2 players and players take turns.
+// 2. When a player clicks Submit, the game rolls 2 dice and shows the dice rolls, for example 3 and 6.
+// 3. The player picks the order of the dice they want. For example, if they wanted the number 63, they would specify that the 2nd dice goes first. You can choose how the player specifies dice order.
+// 4. After both players have rolled and chosen dice order, the player with the higher combined number wins.
+
+//Define global variables
+//Checks if game mode is to roll dice or if user needs to specify order
+var GAME_MODE_ROLL_DICE = `dice Roll`;
+var GAME_MODE_CHOOSE_ORDER = `choose Order`
+var UNIVERSAL_GAME_MODE = GAME_MODE_ROLL_DICE;
+var counter = 0;
+var firstPlayerFinalDiceRoll = 0;
+var secondPlayerFinalDiceRoll = 0;
+var playerDiceRoll = [];
+var errorMessage = ``;
+var chosenOrderOfDiceRoll = 0;
+
+
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
-};
+
+  //loop exits after 2 rounds where player 1 & 2 choose dice values
+  while (counter < 2) {
+  // starts game with dice roll
+  if (UNIVERSAL_GAME_MODE === GAME_MODE_ROLL_DICE) { 
+    var totalDiceRolls = [];
+    totalDiceRolls.push(rollDice());
+    UNIVERSAL_GAME_MODE = GAME_MODE_CHOOSE_ORDER;
+    return totalDiceRolls;
+  }
+  //proceeds to game section to choose dice order
+  if ( UNIVERSAL_GAME_MODE === GAME_MODE_CHOOSE_ORDER ) {
+    chosenOrderOfDiceRoll = validatesInput(input);
+
+    if (chosenOrderOfDiceRoll === errorMessage) {
+      return chosenOrderOfDiceRoll;
+    }
+
+    if (chosenOrderOfDiceRoll !== errorMessage) {
+      var reorderDiceRoll = orderDiceRoll(chosenOrderOfDiceRoll);
+      var finalDiceOrder = capturesDiceRollForEachPlayer(reorderDiceRoll);
+      UNIVERSAL_GAME_MODE = GAME_MODE_ROLL_DICE
+      return finalDiceOrder;
+    }
```

Comment:
if `chosenOrderOfDiceRoll` is equal `errorMessage`, we already return. Whatever follows must naturally mean `chosenOrderOfDiceRoll !== errorMessage` so we do not have to explicitly check for it.
```suggestion
    if (chosenOrderOfDiceRoll === errorMessage) {
      return chosenOrderOfDiceRoll;
    }

    var reorderDiceRoll = orderDiceRoll(chosenOrderOfDiceRoll);
    var finalDiceOrder = capturesDiceRollForEachPlayer(reorderDiceRoll);
    UNIVERSAL_GAME_MODE = GAME_MODE_ROLL_DICE
    return finalDiceOrder;
    
```